### PR TITLE
BXMSPROD-560 lock-treatment-tool added to Maven build

### PR DESCRIPTION
**Pull request by Ginxo**
BXMSPROD-560

### CODE REVIEW

**Comment by Ginxo** [comment]
Jenkins retest this please

**Comment by Christopher-Chianelli** [comment]
Do we only want to run it for the productize profile? I think it might be useful to have it in the normal build. Checks pass + I verified it works on my machine (well, not via this PR, but I pretty much added your changes to something else except I use the default profile)

File: employee-rostering-frontend/pom.xml [hunk 1]
```diff
@@ -29,21 +29,27 @@
   <name>Employee Rostering as a Service React Frontend</name>
   <description>React frontend for Employee Rostering as a Service</description>
 
+  <properties>
+    <frontend-maven-plugin.version>1.7.5</frontend-maven-plugin.version>
+    <node.version>v8.12.0</node.version>
+    <npm.version>6.9.0</npm.version>
+</properties>
```
Comment (Christopher-Chianelli) [inline employee-rostering-frontend/pom.xml]: wrong identation

**Comment by Ginxo** [comment]
@Christopher-Chianelli lock-treatment-tool moved to the default build and indentation fixed

**Comment by Christopher-Chianelli** [comment]
@Ginxo still wrong indentation on line 36. Everything else looks good. Will this set of PR's be merged as one or individually?

**Comment by Ginxo** [comment]
@Christopher-Chianelli fixed, sorry for that

**Comment by mareknovotny** [comment]
Jenkins retest this please

**Comment by Ginxo** [comment]
@mareknovotny moved to default profile.
The same done for optaweb-vehicle-routing https://github.com/kiegroup/optaweb-vehicle-routing/pull/218

**Comment by Ginxo** [comment]
Jenkins retest this please

**Comment by Ginxo** [comment]
@Christopher-Chianelli @mareknovotny I don't see any error in the travis log and build result is success

**Comment by Christopher-Chianelli** [comment]
@Ginxo you need to remove `package-lock.json`, then run `mvn clean install`, then commit the updated `package-lock.json` file (Travis fails if it detects running an install cause package-lock changes and those changes are not commited).

File: employee-rostering-frontend/pom.xml [hunk 2]
```diff
@@ -29,25 +29,53 @@
...
             </goals>
             <configuration>
-              <nodeVersion>v8.12.0</nodeVersion>
-              <npmVersion>6.9.0</npmVersion>
+              <nodeVersion>${node.version}</nodeVersion>
+              <npmVersion>${npm.version}</npmVersion>
             </configuration>
           </execution>
+          <execution>
+              <id>npm install lock-treatment-tool and run-node</id>
```
Comment (yurloc) [inline employee-rostering-frontend/pom.xml]: Wrong indentation. Please keep the formatting consistent.

**Comment by Ginxo** [comment]
I see, two things were happening here:

- the `--no-save` flag absence when installing lock-treatment-tool. This was adding the library to dependencies and as a result the package-lock.json changes once the `install` is executed.
- The `mvn generate-sources` command executed again the 'lock-treatment-tool execution' execution step. Moved to 'compile' phase which I don't like but it's the only way (I'm able to see) to skip it.

@yurloc I think I've solved my indentation problems for the rest of my life ;) (I hope so)
And about your question, @mareknovotny proposed my to keep all the projects aligned and the default profile is the chosen one.

**Comment by Christopher-Chianelli** [comment]
@yurloc another advantage of the lock-treatment tool being executed in the default profile: the lock file will no longer contain urls; only version numbers and integrity.

**Comment by Christopher-Chianelli** [comment]
@Ginxo I recommend updating `package-lock.json`, since it will be updated anyways whenever we add a new dependency (note: old entries will have their URL's removed)

**Comment by Ginxo** [comment]
@Christopher-Chianelli the package-lock.json is up to date since the lock-treatment-tool library is installed - -global-style and this is the way we should. 
Travis green

**Comment by Christopher-Chianelli** [comment]
I am not sure if it is working now; I did a `mvn clean install` with these commit applied and the url were still added to the package-lock.json; I needed to run `npm run env -- locktt` after the install to remove those fields. We might want to move that execution to the `package` phase.

**Pull request merged**